Update and expand Build System feature module documentation (#887)
spatialos/gdk-for-unity: docs/modules/build-system/editor-menu.md

# Editor menu

> The Build System Feature Module also provides a set of Unity Editor menu items to build your workers from your Unity Editor. These can be accessed via the **SpatialOS** menu item. 

Reviewer: ```suggestion
The Build System Feature Module provides a set of Unity Editor menu items to build your workers from your Unity Editor. These can be accessed via the **SpatialOS** menu item. 
```

Reviewer: marked as resolved, but not changed :P

Author: I thought I did? 🤔

> You can build for a specific target: **Build for local** and **Build for cloud** and then by specific worker type, or build all workers.

Reviewer: ```suggestion
You can build for a specific target: **Build for local** and **Build for cloud** and then either select a specific worker type or build all workers for this target.
```

> For example, to build `UnityClient` for a cloud deployment you would select **SpatialOS** > **Build for cloud** > **UnityClient**.

Reviewer: ```suggestion
For example, to build the worker type `UnityClient` for a cloud deployment you would select **SpatialOS** > **Build for cloud** > **UnityClient**.
```
